core/ssl.c

```diff
@@ -30,7 +30,18 @@ void uwsgi_ssl_info_cb(SSL const *ssl, int where, int ret) {
 }
 
 int uwsgi_ssl_verify_callback(int ok, X509_STORE_CTX * x509_store) {
-        return 1;
+        char buf[256];
+        X509 *err_cert;
+        int depth;
+        int err;
+        depth = X509_STORE_CTX_get_error_depth(x509_store);
+        err_cert = X509_STORE_CTX_get_current_cert(x509_store);
+        X509_NAME_oneline(X509_get_subject_name(err_cert), buf, 256);
+        err = X509_STORE_CTX_get_error(x509_store);
+        if (!ok) {
```

Comment:
Do you mean outside this function ? Full list of verification errors is accessible only from this callback. We can move 'uwsgi_log' outside but still will be forced to store error data in some context. Another question - where to extract error data back from this context. I see only one possibility in this case - store special flag that we have ssl enabled and extract errors after *->proto_accept call.

Also there is another possible solution - move logging under UWSGI_DEBUG macro.